src/main/java/de/tum/in/www1/artemis/domain/notification/GroupNotificationFactory.java

```diff
@@ -165,57 +194,92 @@
     public static GroupNotification createNotification(Post post, User author, GroupNotificationType groupNotificationType, NotificationType notificationType, Course course) {
         String title;
         String text;
+        String[] placeholderValues;
         GroupNotification notification;
         switch (notificationType) {
             case NEW_EXERCISE_POST -> {
```

Comment:
I think the common placeholder values should always start the array. This way we can keep the logic for getting course title and post stuff identical for all message types on the native clients. The exercise title/lecture title etc should be appended then.

Reply:
Didn't we say that the placeholder array need to be key-based since the order can change in different languages? That would also fix your comment

Reply:
I thought we agreed index based is fine. I changed the order according to Tims's suggestion.